prime-router/src/main/kotlin/Submission.kt

```diff
@@ -76,6 +78,114 @@ class DetailedSubmissionHistory(
      */
     var externalName: String? = null
 
+    /**
+     * The step in the delivery process for a submission
+     * Supported values:
+     *     ERROR - error on initial submission
+     *     RECEIVED - passed the received step in the pipeline and awaits processing/routing
+     *     NOT_DELIVERING - processed but has no intended receivers
+     *     WAITING_TO_DELIVER - processed but yet to be sent to/downloaded by any receivers
+     *     PARTIALLY_DELIVERED - processed, successfully sent to/downloaded by at least one receiver
+     *     DELIVERED - processed, successfully sent to/downloaded by all receivers
+     * @todo For now, no "send error" type of state.
+     *     If a send error occurs for example,
+     *     it'll just sit in the waitingToDeliver or
+     *     partiallyDelivered state until someone fixes it.
+     */
+    enum class Status(val printableName: String) {
+        ERROR("Error"),
+        RECEIVED("Received"),
+        NOT_DELIVERING("Not Delivering"),
+        WAITING_TO_DELIVER("Waiting to Deliver"),
+        PARTIALLY_DELIVERED("Partially Delivered"),
+        DELIVERED("Delivered"),
+    }
+
+    /**
+     * Summary of how far along the submission's process is.
+     * The supported values arre listed in the Status enum.
+     */
+    val overallStatus: Status get() {
+        if (httpStatus != HttpStatus.OK.value() && httpStatus != HttpStatus.CREATED.value()) {
+            return Status.ERROR
+        }
+
+        if (destinations.size == 0) {
+            // 0 unfiltered destinations = unprocessed in almost every situation
```

Comment:
Flesh out this comment more.   Like:

There are two cases that might cause destinations.size == 0:
1.  Data hasn't been 'processed' yet. 
2.  Very rare:  No data matches any geographical location.  For example, if both the testing tab and patient data were foreign addresses.
At the moment we have NO easy way to distinguish the latter rare case, so it will be treated as status `RECEIVED` as well.